From 0fc2da8d1129f6daaff0f9f0be96b73e3bd5e49e Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Wed, 22 Mar 2023 18:41:17 +0800
Subject: [PATCH 1/2] nnstreamer: Fix include header, header in tensorflow
 directory, not tensorflow2 directory

Build error due to incorrect haeder path. Header is in /usr/include/tensorflow/lite,
not /usr/include/tensorflow2/lite.

Signed-off-by: kidd-kw.chen <kidd-kw.chen@mediatek.com>
---
 ext/nnstreamer/tensor_filter/meson.build             |  2 +-
 .../tensor_filter/tensor_filter_tensorflow_lite.cc   | 12 ++++++------
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/ext/nnstreamer/tensor_filter/meson.build b/ext/nnstreamer/tensor_filter/meson.build
index dd54094b..9e6196f8 100644
--- a/ext/nnstreamer/tensor_filter/meson.build
+++ b/ext/nnstreamer/tensor_filter/meson.build
@@ -194,7 +194,7 @@ if tflite2_support_is_available
   tflite2_compile_args += '-DTFLITE_FLOAT16=1'
   tflite2_compile_args += '-DTFLITE_COMPLEX64=1'
 
-  if cc.has_header('tensorflow2/lite/model.h')
+  if cc.has_header('tensorflow/lite/model.h')
     tflite2_compile_args += '-DUSE_TENSORFLOW2_HEADER_PATH=1'
   endif
 
diff --git a/ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc b/ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc
index d7693e52..8d62ffaa 100644
--- a/ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc
+++ b/ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc
@@ -41,8 +41,8 @@
 
 #if TFLITE_VERSION_MAJOR >= 2 || TFLITE_VERSION_MINOR >= 13
 #  if USE_TENSORFLOW2_HEADER_PATH
-#    include <tensorflow2/lite/kernels/register.h>
-#    include <tensorflow2/lite/model.h>
+#    include <tensorflow/lite/kernels/register.h>
+#    include <tensorflow/lite/model.h>
 #  else
 #    include <tensorflow/lite/kernels/register.h>
 #    include <tensorflow/lite/model.h>
@@ -55,7 +55,7 @@
 /** control delegate headers */
 #ifdef TFLITE_XNNPACK_DELEGATE_SUPPORTED
 #  if USE_TENSORFLOW2_HEADER_PATH
-#    include <tensorflow2/lite/delegates/xnnpack/xnnpack_delegate.h>
+#    include <tensorflow/lite/delegates/xnnpack/xnnpack_delegate.h>
 #  else
 #    include <tensorflow/lite/delegates/xnnpack/xnnpack_delegate.h>
 #  endif
@@ -63,7 +63,7 @@
 
 #ifdef TFLITE_GPU_DELEGATE_SUPPORTED
 #  if USE_TENSORFLOW2_HEADER_PATH
-#    include <tensorflow2/lite/delegates/gpu/delegate.h>
+#    include <tensorflow/lite/delegates/gpu/delegate.h>
 #  else
 #    include <tensorflow/lite/delegates/gpu/delegate.h>
 #  endif
@@ -71,7 +71,7 @@
 
 #ifdef TFLITE_NNAPI_DELEGATE_SUPPORTED
 #  if USE_TENSORFLOW2_HEADER_PATH
-#    include <tensorflow2/lite/delegates/nnapi/nnapi_delegate.h>
+#    include <tensorflow/lite/delegates/nnapi/nnapi_delegate.h>
 #  else
 #    include <tensorflow/lite/delegates/nnapi/nnapi_delegate.h>
 #  endif
@@ -79,7 +79,7 @@
 
 #ifdef TFLITE_EXTERNAL_DELEGATE_SUPPORTED
 #  if USE_TENSORFLOW2_HEADER_PATH
-#    include <tensorflow2/lite/delegates/external/external_delegate.h>
+#    include <tensorflow/lite/delegates/external/external_delegate.h>
 #  else
 #    include <tensorflow/lite/delegates/external/external_delegate.h>
 #  endif
-- 
2.18.0

